packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -13,33 +12,20 @@ type Props = CommonInteractionProps;
 
 function SelectToZoom(props: Props) {
   const context = useAxisSystemContext();
-  const { visRatio } = context;
+  const { canvasSize, canvasRatio, visRatio } = context;
+  const { width, height } = canvasSize;
   const keepRatio = visRatio !== undefined;
 
-  const moveCameraTo = useMoveCameraTo();
-
-  const { width, height } = useThree((state) => state.size);
   const camera = useThree((state) => state.camera);
-
-  function getDataRatio() {
-    const { xVisibleDomain, yVisibleDomain } = getVisibleDomains(
-      camera,
-      context
-    );
-    return Math.abs(
-      (xVisibleDomain[1] - xVisibleDomain[0]) /
-        (yVisibleDomain[1] - yVisibleDomain[0])
-    );
-  }
+  const moveCameraTo = useMoveCameraTo();
 
   function onSelectionEnd(selection: Selection) {
     const { startPoint: dataStartPoint, endPoint: dataEndPoint } = selection;
-    const dataRatio = getDataRatio();
 
     // Work in world coordinates as we need to act on the world camera
     const [startPoint, endPoint] = (
       keepRatio
-        ? getRatioRespectingRectangle(dataStartPoint, dataEndPoint, dataRatio)
+        ? getRatioRespectingRectangle(dataStartPoint, dataEndPoint, canvasRatio)
```

Comment:
Looking back at these, we might be able to simplify this by allowing `getRatioRespectingRectangle` to accept an `undefined` ratio that triggers an early return.

Reply:
Excellent idea. I've also re-used the `.map(scale)` pattern in `RatioSelectionRect`.